### PR TITLE
fix(streaming): ensure schema and order are consistent in lookup

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

Currently, all TPC-H can pass under delta join. (Only tested single node).

image

This PR fixes a lot of bugs. Now we ensures that:

* Lookup will output `stream + arrangement` internally, and then reordered by column reordering parameters.
* Lookup's input must be `stream` then `arrangement`.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

blocked by https://github.com/singularity-data/risingwave/issues/1901, causing MV with delta join cannot be dropped.
close https://github.com/singularity-data/risingwave/issues/1882

### CODE REVIEW

**Comment by skyzh** [comment]
also hold until https://github.com/singularity-data/risingwave/issues/1967? Modification on the schema part will easily break some schema checks. Should wait until there's a schema check executor so that we can be sure that there's no schema mismatch problems.

**Comment by BugenZhao** [comment]
> also hold until #1967? Modification on the schema part will easily break some schema checks. Should wait until there's a schema check executor so that we can be sure that there's no schema mismatch problems.

Good idea. I believe it will be quick.

**Review by fuyufjh — APPROVED** [review]
LGTM.

By the way, IIUC, there is no e2e test for looking up shared state now?

**Comment by fuyufjh** [comment]
> By the way, IIUC, there is no e2e test for looking up shared state now?

Oh it's covered in #1978

File: src/stream/src/executor_v2/lookup/sides.rs [hunk 1]
```diff
@@ -61,11 +61,16 @@ pub(crate) struct ArrangeJoinSide<S: StateStore> {
     /// The column descriptors of columns in arrangement
     pub col_descs: Vec<ColumnDesc>,
 
-    /// Order rules of the arrangement (used for lookup)
+    /// Order rules of the arrangement (only join key is needed, pk should not be included, used
+    /// for lookup)
     pub order_rules: Vec<OrderPair>,
```
Comment (yuhao-su) [inline src/stream/src/executor_v2/lookup/sides.rs:66]: I saw a definition of `arrangement_order_rules` in `LookupExecutorParams` that mentioned the primary key of MV.  And the `order_rules` here is created from `arrangement_order_rules`. So what is the difference between those two order_rules?
```rust
    /// Should be the same as [`OrderPair`] in the arrangement.
    ///
    /// Still using the above `a, b, _row_id` example. If we create an arrangement with join key
    /// `a`, there will be two elements in `arrangement_order_rules`.
    ///
    /// * The first element is the order rule for `a`, which is the join key. Join keys should
    ///   always come first.
    /// * The second element is the original primary key for the MV.
    ///
    /// These two parts now compose the arrange keys for the arrangement.
    pub arrangement_order_rules: Vec<OrderPair>,
```

Comment (skyzh) [reply src/stream/src/executor_v2/lookup/sides.rs:66]: `arrangement_order_rules` and `order_rules` are exactly the same.

**Comment by yuhao-su** [comment]
LGTM

**Review by yuhao-su — APPROVED** [review]

**Pull request merged**